### PR TITLE
first working draft for chat cooldown system

### PR DESCRIPTION
**Pull request by druu**
@ryanwinchester it works. but I'm sure you want to have a look at it :D

### CODE REVIEW

File: lib/smwc_bot/fridge.ex [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    map_to_sha256(map)
+    |> remove()
+  end
+
+  def remove(hash) when is_bitstring(hash) do
+    if Map.has_key?(get_contents(), hash) do
+      GenServer.call(__MODULE__, {:delete, hash})
+    else
+      true
+    end
```
Comment (ryanwinchester) [inline lib/smwc_bot/fridge.ex:76]: do you need an `else`?

File: lib/smwc_bot/fridge.ex [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  end
+
+  defp map_to_sha256(map) do
+    map
+    |> Map.to_list()
+    |> Enum.reduce("", fn {k,v}, acc -> "#{k}#{v}" <> acc end)
+    |> string_to_sha256()
```
Comment (ryanwinchester) [inline lib/smwc_bot/fridge.ex:83]: I think string-building is more efficient if you make the list and then just do `IO.iodata_to_binary()`

File: lib/smwc_bot/fridge.ex [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  end
```
Comment (ryanwinchester) [inline lib/smwc_bot/fridge.ex:84]: I think string-building is more efficient if you make the list and then just do `IO.iodata_to_binary()`

```suggest
  defp map_to_sha256(map) do
    map
    |> Enum.map(&Tuple.to_list/1)
    |> IO.iodata_to_binary()
    |> string_to_sha256()
  end
```

File: lib/smwc_bot/fridge.ex [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  defp string_to_sha256(str) do
+    :crypto.hash(:sha256, str)
+    |> Base.encode16()
```
Comment (ryanwinchester) [inline lib/smwc_bot/fridge.ex:88]:
```elixir
Base.encode16(str, case: :lower)
```

File: lib/smwc_bot/fridge.ex (hunk 1 above)
Comment (druu) [reply lib/smwc_bot/fridge.ex:76]: removed. don't care about a return value

File: lib/smwc_bot/fridge.ex (hunk 3 above)
Comment (druu) [reply lib/smwc_bot/fridge.ex:84]: Taking your word for it here. had to make sure map/2 returns [string, string]

File: lib/smwc_bot/fridge.ex (hunk 4 above)
Comment (druu) [reply lib/smwc_bot/fridge.ex:88]: done

**Pull request merged**